### PR TITLE
Native LibreWolf compatibility

### PR DESCRIPTION
**Pull request by ghost**
This change adds native compatibility for the LibreWolf Browser.

### CODE REVIEW

File: src/browser/NativeMessageInstaller.cpp [hunk 1]
```diff
@@ -58,6 +59,8 @@ namespace
         QStringLiteral("HKEY_CURRENT_USER\\Software\\Chromium\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser");
     const QString TARGET_DIR_FIREFOX =
         QStringLiteral("HKEY_CURRENT_USER\\Software\\Mozilla\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser");
+    const QString TARGET_DIR_LIBREWOLF =
+        QStringLiteral("HKEY_CURRENT_USER\\Software\\LibreWolf\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser");
```
Comment (droidmonkey) [inline src/browser/NativeMessageInstaller.cpp:63]: Should just set this equal to the FIREFOX variable.

Comment (LetMeByte) [reply src/browser/NativeMessageInstaller.cpp:63]: Please elaborate?

Comment (LetMeByte) [reply src/browser/NativeMessageInstaller.cpp]: I don't see the technical reason for them to be the same, but I changed it back. Will that be everything required to merge?

**Comment by LetMeByte** [comment]
I don't see the technical reason for them to be the same, but I changed it back. Will that be everything required to merge?

File: src/browser/NativeMessageInstaller.cpp [hunk 2]
```diff
@@ -222,6 +230,8 @@ QString NativeMessageInstaller::getNativeMessagePath(SupportedBrowsers browser)
         basePath = QDir::homePath() + "/.local/share";
     } else if (browser == SupportedBrowsers::FIREFOX) {
         basePath = QDir::homePath();
+    } else if (browser == SupportedBrowsers::LIBREWOLF) {
```
Comment (varjolintu) [inline src/browser/NativeMessageInstaller.cpp]: Use OR.

File: src/browser/NativeMessageInstaller.cpp [hunk 3]
```diff
@@ -58,6 +59,8 @@ namespace
         QStringLiteral("HKEY_CURRENT_USER\\Software\\Chromium\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser");
     const QString TARGET_DIR_FIREFOX =
         QStringLiteral("HKEY_CURRENT_USER\\Software\\Mozilla\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser");
+    const QString TARGET_DIR_LIBREWOLF =
+        QStringLiteral("HKEY_CURRENT_USER\\Software\\Mozilla\\NativeMessagingHosts\\org.keepassxc.keepassxc_browser");
```
Comment (varjolintu) [inline src/browser/NativeMessageInstaller.cpp]: LibreWolf really uses Mozilla's path on Windows?

File: src/browser/NativeMessageInstaller.cpp [hunk 4]
```diff
@@ -230,6 +240,8 @@ QString NativeMessageInstaller::getNativeMessagePath(SupportedBrowsers browser)
         basePath = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation);
     } else if (browser == SupportedBrowsers::FIREFOX) {
         basePath = QDir::homePath();
+    } else if (browser == SupportedBrowsers::LIBREWOLF) {
```
Comment (varjolintu) [inline src/browser/NativeMessageInstaller.cpp]: Use OR.

File: src/browser/NativeMessageInstaller.cpp (hunk 3 above)
Comment (ghost) [reply src/browser/NativeMessageInstaller.cpp]: Yes, I tested it. Works out of the box and also works while FF and LW are both using the addons on the same DB at the same time.

I set a custom path before, but droidmonkey told me to change it back to the same path as for Firefox, so I did that.
I made sure to ask why that's not the right way, but then his comment vanished.

The LibreWolf team already has a patch to use the unique path, so it would also work with that.
Wouldn't the resulting JSON file be in the same path, anyway?

File: src/browser/NativeMessageInstaller.cpp (hunk 2 above)
Comment (LetMeByte) [reply src/browser/NativeMessageInstaller.cpp]: done, thanks for the advice.

File: src/browser/NativeMessageInstaller.cpp (hunk 3 above)
Comment (droidmonkey) [reply src/browser/NativeMessageInstaller.cpp]: I didn't say to change it back, you had originally made the PR with Mozilla. I told you to assign the LIBREWOLF variable to the Mozilla one to make it explicit that they were the same path.

Comment (ghost) [reply src/browser/NativeMessageInstaller.cpp]: Then I misunderstood your comment; at the time you commented, I already made the path unique. At first I used the same path, because that also works, but the LW team decided a unique path would be better.

TLDR; can we use the unique path then?

Comment (droidmonkey) [reply src/browser/NativeMessageInstaller.cpp]: If it works, then yes absolutely.

Comment (ghost) [reply src/browser/NativeMessageInstaller.cpp]: Thank you for the approval, I'm sure it will work. The json file will still be the same, at least that's how I understand it, so there shouldn't be any problem.

**Comment by ghost** [comment]
Can this be merged? I made all the necessary changes.

**Comment by droidmonkey** [comment]
Code format is wrong, need to run `make format`

**Comment by ghost** [comment]
@droidmonkey It built.

File: src/browser/NativeMessageInstaller.cpp [hunk 5]
```diff
@@ -220,15 +228,17 @@ QString NativeMessageInstaller::getNativeMessagePath(SupportedBrowsers browser)
...
+    } else if (browser == SupportedBrowsers::FIREFOX || browser == SupportedBrowsers::LIBREWOLF)
         basePath = QDir::homePath();
-    } else {
-        basePath = QDir::homePath() + "/.config";
-    }
+}
+else
+{
+    basePath = QDir::homePath() + "/.config";
+}
```
Comment (michaelk83) [inline src/browser/NativeMessageInstaller.cpp]: I don't understand how this passes code formatting? This is clearly wrong indentation (and line breaks) here, and inconsistent with the rest of the code...

Comment (ghost) [reply src/browser/NativeMessageInstaller.cpp]: I wondered about that, too. It was good before I ran the code format, I made sure to format everything correctly by hand. But that caused problems with the checks.

EDIT: Fixed the formatting, now the check fails again. Something's off here.

**Comment by ghost** [comment]
This doesn't look like something I could fix. How do we resolve this?

**Comment by ghost** [comment]
Here's what's happening:

1. The formatting is right, but the GH formatting check fails.
2. Using the 'make format' tool actually messes up the indentation, but GH says it's right.

To me this looks like the code formatter and checker seem to be broken. Please merge once you fixed it, or tell me what to do.

**Review by ghost — COMMENTED** [review]
Merge now or never.
